src/security.c

```
@@ -1,1012 +1,616 @@
-
-        platform_flash_write(COUNTER_PRIMARY_ADDR, &security_state.primary_counter, sizeof(monotonic_counter_t));
+    /* We are responder: received HELLO */
+    if (msg->type == HANDSHAKE_TYPE_HELLO && msg->responder_id == our_node_id) {
+        if (sodium_is_zero(msg->public_key, PUBLIC_KEY_LENGTH)) {
+            return -2;
+        }
 
-        /* Initialize backup too */
-        security_state.backup_counter = security_state.primary_counter;
-        platform_flash_write(COUNTER_BACKUP_ADDR, &security_state.backup_counter, sizeof(monotonic_counter_t));
+        uint8_t shared[SHARED_SECRET_LENGTH];
```

Clang (14.0.6)

[warning] variable 'shared' is not initialized

(cppcoreguidelines-init-variables)